### PR TITLE
Rename SetGasLimit -> SetComputeLimit

### PR DESCRIPTION
**Pull request by nozim**
Closes: #291 

## Description

______

For contributor use:

- [x] Targeted PR against `master` branch
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [x] Code follows the [standards mentioned here](https://github.com/onflow/flow-go-sdk/blob/master/CONTRIBUTING.md#styleguides).
- [x] Updated relevant documentation 
- [x] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Added appropriate labels

### CODE REVIEW

**Comment by sideninja** [comment]
@chasefleming mind reviewing?

**Review by nvdtf — APPROVED** [review]

**Comment by nvdtf** [comment]
There's a page to update in the docs here https://developers.flow.com/concepts/start-here/variable-transaction-fees#configuring-execution-limits

**Comment by nozim** [comment]
@nvdtf https://github.com/onflow/docs/pull/206

**Comment by nozim** [comment]
@chasefleming @janezpodhostnik  kindly help review 🙏

File: transaction.go [hunk 1]
```diff
@@ -182,8 +183,8 @@ func (t *Transaction) SetReferenceBlockID(blockID Identifier) *Transaction {
 	return t
 }
 
-// SetGasLimit sets the gas limit for this transaction.
-func (t *Transaction) SetGasLimit(limit uint64) *Transaction {
+// SetComputeLimit sets the gas limit for this transaction.
```
Comment (janezpodhostnik) [inline transaction.go:186]:
```suggestion
// SetComputeLimit sets the compute limit for this transaction.
```

**Review by janezpodhostnik — CHANGES_REQUESTED** [review]
Thank you for the rename!

can you also check if you find any text references of "gas" in the comments or the readmes of this repo?

**Comment by nozim** [comment]
@janezpodhostnik thanks, will do :)

File: transaction.go (hunk 1 above)
Comment (nozim) [reply transaction.go]: https://github.com/onflow/flow-go-sdk/pull/448/commits/e78be7212693c76d49638d68194f603716152485

There is still "gas" in protobuf and other places and will involve changing other places as well.

**Comment by nozim** [comment]
@janezpodhostnik updated. Kindly re-review :)

**Comment by turbolent** [comment]
Maybe the new function should be added, and the old one should be marked as deprecated, kept for a while, and eventually removed after some reasonable time?

**Comment by janezpodhostnik** [comment]
@turbolent good point.

@nozim lets keep the original method in `transaction.go` but mark it as deprecated:

```go
// SetGasLimit sets the gas limit for this transaction.
//
// Deprecated: Use SetComputationLimit Instead
func (t *Transaction) SetGasLimit(limit uint64) *Transaction {
	t.GasLimit = limit
	return t
}
```

Everything else lgtm.

**Review by janezpodhostnik — APPROVED** [review]

**Pull request merged**